logstash.go

```diff
@@ -54,17 +35,32 @@ func NewLogstashAdapter(route *router.Route) (router.LogAdapter, error) {
 	}
 
 	return &LogstashAdapter{
-		route: route,
-		conn:  conn,
+		route:         route,
+		conn:          conn,
+		containerTags: make(map[string][]string),
 	}, nil
 }
 
+// Get container tags configured with the environment variable LOGSTASH_TAGS
+func GetContainerTags(c *docker.Container, a *LogstashAdapter) []string {
+	var tags []string
+	if val, ok := a.containerTags[c.ID]; ok {
+		tags = val
+	} else {
+		for _, e := range c.Config.Env {
+			if strings.HasPrefix(e, "LOGSTASH_TAGS=") {
+				tags = strings.Split(strings.TrimPrefix(e, "LOGSTASH_TAGS="), ",")
+				break
+			}
+		}
+		a.containerTags[c.ID] = tags
+	}
+	return tags
```

Comment:
Ah, got it. Sure, that will work fine!

On mån 29 aug. 2016 at 22:45, Ricardo Reis <notifications@github.com> wrote:

> In logstash.go
> <https://github.com/looplab/logspout-logstash/pull/31#discussion_r76683496>
> :
>
> >  	}, nil
> >  }
> >
> > +// Get container tags configured with the environment variable LOGSTASH_TAGS
> > +func GetContainerTags(c *docker.Container, a *LogstashAdapter) []string {
> > +	var tags []string
> > +	if val, ok := a.containerTags[c.ID]; ok {
> > +		tags = val
> > +	} else {
> > +		for _, e := range c.Config.Env {
> > +			if strings.HasPrefix(e, "LOGSTASH_TAGS=") {
> > +				tags = strings.Split(strings.TrimPrefix(e, "LOGSTASH_TAGS="), ",")
> > +				break
> > +			}
> > +		}
> > +		a.containerTags[c.ID] = tags
> > +	}
> > +	return tags
>
> I still need to add the key here, else for a container with no tags it
> always loop through all the environment variables... What do you think of
> declaring the variable before the for loop and keep the break statement?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/looplab/logspout-logstash/pull/31/files/ca79e077e868c55d5f290c61d92ec7c9e7b6909d#r76683496>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAyJDkH_HvVgsslrW0XEKaLhmSunrR76ks5qk0TRgaJpZM4JviuK>
> .
>
-- 
Max Ekman
Software Engineer

+46 708 710504
max@looplab.se

＞ looplab.se